app/models/user.rb

```diff
@@ -71,6 +71,27 @@ class User < ApplicationRecord
   scope :with_email, ->(email) { where('lower(email) = ?', email.downcase) }
 
   before_save :capitalize_name_fragments
+  before_save :record_name_change, if: :will_save_change_to_name?
```

Comment:
We will also need to log the user who did the action; It would be good to move it into a service and call it from admin and profile